### ISSUE TITLE
0.6.4 release plan

### ISSUE THREAD

**Issue by chchwy** [pencil2d/pencil#1215]
@scribblemaniac @CandyFace @Jose-Moreno @davidlamhauge 
Let's discuss the v0.6.4 release here.

Since I am not able to find much time working on Pencil2D lately, I will need you guys' help to release v0.6.4.

Firstly, please review the milestone. Add the high-priority issues & PR to the milestone and push back the low priority issues to the next version.
https://github.com/pencil2d/pencil/milestone/6

Secondly, we need a volunteer release coordinator. He will be responsible for validating fixes, preparing release note, changelog. And of course, make the call on releasing v0.6.4.

Luckily we don't need an emergency fix in v0.6.4 like what we had in 0.6.1 and 0.6.2. So it will be a general maintenance release. (maybe include some features?)

**Comment by scribblemaniac** [comment]
The only open issue that I think is important to fix before the release is #1195. Some of the other ones are equally or even more important, but there is nothing critical, and nothing else that we can fix easily. I believe that PRs #1213, #1211, and parts of #1214 should be merged before the release as well. I am against adding any features, that's just asking for trouble.

I will take up coordination of this release if there are no objections or other people who wish to do it. Someone else should validate the fixes however as the majority of them are mine or have already been reviewed by me.

**Comment by CandyFace** [comment]
Agreed, fixing the remaining issues in the milestone is going to be tough as non of the remaining ones are easy to fix, that is... if we want to release soon.

I fully support you taking charge of the coordination @scribblemaniac.
I've code reviewed all the PR's with bug fixes (except the recent #1214) but I have not compiled and validated that nothing is broken.

**Comment by Jose-Moreno** [comment]
I also endorse the idea of @scribblemaniac being appointed as release coordinator. Additionally I can provide support with the release notes and changelog like I did last time. No problem.

I agree that we should be careful not to introduce new features as per our team discussion 0.6.4 would be focused on bug fixing. Some of the milestone elements are actually enhancement requests which we can kick to 0.6.5 or 0.6.7 which would be "feature releases" (but also would depend on the complexity of their implementation)

All in all this is basically a house-keeping release to stay true to our (give and take) 3-month release cycle. Let me know what else I can do to assist you all.

**Comment by scribblemaniac** [comment]
Here's a copy of the release checklist that we'll use to track the release of this version:

#### Source code
- [ ] Bump version number (common.pri & info.plist)
- [ ] Prepare release notes for CHANGELOG
- [ ] Disable debug information
- [ ] Make sure all unit tests are working and are OK
- [ ] Test all features known to work and make sure they behave the same
- [ ] Update contributor list
- [ ] Make sure 0 issues in milestone
- [ ] Add release tag to the correct commit
- [ ] Push to release branch

#### Deployment
- [ ] Build and upload 32-bit Linux AppImage
- [ ] Add downloads to the [Bitbucket mirror](https://bitbucket.org/chchwy/pencil2d)
- [ ] Update download links:
  - [ ] [Pencil2d.org](https://www.pencil2d.org/download/)
  - [ ] [Github README](https://github.com/pencil2d/pencil)
- [ ] Update packages
  - [ ] AUR (Arch Linux) @J5lx
  - [ ] Homebrew Cask (macOS) @scribblemaniac
  - [ ] Apt (Debian/Ubuntu) @mapreri
  - [ ] Flatpak ([link](https://github.com/pencil2d/pencil/issues/1036)) @eszlari
  - [ ] RPM (Fedora) @blackfile
  - [ ] FreeBSD @yurivict

#### Social media
Make announcement with release notes (or a link to the release notes) on the following platforms:
- [ ] [Pencil2d.org News](https://www.pencil2d.org/news.html)
- [ ] [Facebook](https://www.facebook.com/groups/709024775972087/)
- [ ] [Discord](https://discord.gg/8FxdV2g)
- [ ] [Google+](https://plus.google.com/communities/101311036131376070915)
- [ ] [Tumblr](https://www.tumblr.com/tagged/pencil2d)
- [ ] [Youtube?](https://www.youtube.com/channel/UCexMjl4rq8OyzAc7C03uKCQ)

**Comment by mapreri** [comment]
looks good.

Do you think you could update the README _before_ doing the release so that the released tarball has the updated links?  They don't really need to exist while writing the readme, since they are just versioned files, etc.

**Comment by Jose-Moreno** [comment]
@mapreri Thanks for the comment. We still have 1 month and a half to hit the release hard due date that was agreed earlier this year, so Pencil2D could conform to a 3 month release cycle, as well as a slightly different version scheme, which proposed by the team during [January's Dev Meeting.](https://github.com/pencil2d/pencil/wiki/Transcription)

For now we're just laying the groundwork to avoid the same thing happening as before, so please keep track of this thread and the ["version bump" PR](https://github.com/pencil2d/pencil/pull/1217) and once the checklist is fulfilled it will be time to release :smile:

**Comment by scribblemaniac** [comment]
@mapreri Yes good idea. I will amend #1217 to update the README right before the release.

**Comment by scribblemaniac** [comment]
I just updated the translations again. Now that #1220 is resolved, I believe we are finally ready to do the release. 8f42d46e19628d6b69b47f1ce1e00503e1e7c8e0 will the the last commit for v0.6.4, excluding the changelog and version bumps. I'm going to start merging those and pushing to the release branch right away.

**Comment by Jose-Moreno** [comment]
I've updated the changelog to reflect both the closed issue and last commit mentioned. If you see any mistake in the information presented feel free to commit to the [corresponding PR](https://github.com/pencil2d/pencil/pull/1219)

**Comment by scribblemaniac** [comment]
It was a rough road, but I finally got it: the release is now officially live. Now all that remains is to announce the release through social media channels and to update the packages.

P.S. I'm also going to try and build a legacy mac build for this release, but that will take a bit longer and I'm not making any promises.

**Comment by Jose-Moreno** [comment]
@scribblemaniac Thank you for all the hard work. I'll prepare the announcements and go through each during Monday. If there's any last minute changes that need to be done let me know.

**Comment by scribblemaniac** [comment]
An update here: I successfully managed to create a legacy mac build, so this and all future Pencil2D releases will have a legacy mac release that supports Mac OS X 10.7-10.11 (inclusive) 🎉 . For now it will be built manually just like i386 Linux build. Modifications still need to be made to include a link to this build on the downloads page, README, etc. but it can be downloaded now from the Github release binaries. It has been tested briefly on a Mac OS X Lion VM and appears to be fully functional (including movie export after a lot of extra work). I would appreciate any testing that anyone wants to do, especially on versions 10.8+. The ffmpeg plugin for this version does not work on my 10.13 computer, so right now I'm just hoping that it doesn't break until sometime after 10.11 😝 

From what I can tell, we are still waiting for the AUR package to be updated (@J5lx) and the apt package (@mapreri). Just a friendly reminder that we have finished the release now 🙂

**Comment by chchwy** [comment]
Well done @scribblemaniac !
I will manage to make a legacy windows build for Win XP & Vista as well.

**Comment by mapreri** [comment]
Thanks for the reminder!

> Modifications still need to be made to include a link to this build on the downloads page, README, etc.

How so?  The README, as well as the ChangeLog.md, in the released tarball, contain links to 0.6.4, so everything looks good to me.

I have the package for debian ready to be uploaded here, will do soon :)

**Comment by scribblemaniac** [comment]
> How so? The README, as well as the ChangeLog.md, in the released tarball, contain links to 0.6.4, so everything looks good to me.

Sorry for the lack of clarity, I meant that only for the legacy mac build which is not in the README because it was not available at the time of release.

**Comment by mapreri** [comment]
I uploaded 0.6.4 to debian now :)

**Comment by blackfile** [comment]
The Release 0.6.4 is now available on Fedora

**Comment by scribblemaniac** [comment]
Pencil2D has been updated to v0.6.4 on all distros now, and the new legacy builds have been released. I am closing this issue.
